entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md

```diff
@@ -3076,3 +3076,243 @@ LEFT JOIN (
+
+### Unneeded transactions are eliminated
+
+All modern relational databases guarantee transactionality for (most) single SQL statements. That is, the statement will never be only partially completed, even if an error occurs. EF7 avoids starting an explicit transaction in these cases.
+
+For example, looking at the logging for the following call to `SaveChanges`:
+
+<!--
+            await context.AddAsync(new Blog { Name = "MyBlog" });
+            await context.SaveChangesAsync();
+-->
+[!code-csharp[SimpleInsert](../../../../samples/core/Miscellaneous/NewInEFCore7/SaveChangesPerformanceSample.cs?name=SimpleInsert)]
+
+Shows that in EF Core 6.0, the `INSERT` command is wrapped by commands to begin and and then commit a transaction:
+
+```output
+dbug: 9/29/2022 11:43:09.196 RelationalEventId.TransactionStarted[20200] (Microsoft.EntityFrameworkCore.Database.Transaction)
+      Began transaction with isolation level 'ReadCommitted'.
+info: 9/29/2022 11:43:09.265 RelationalEventId.CommandExecuted[20101] (Microsoft.EntityFrameworkCore.Database.Command)
+      Executed DbCommand (27ms) [Parameters=[@p0='MyBlog' (Nullable = false) (Size = 4000)], CommandType='Text', CommandTimeout='30']
+      SET NOCOUNT ON;
+      INSERT INTO [Blogs] ([Name])
+      VALUES (@p0);
+      SELECT [Id]
+      FROM [Blogs]
+      WHERE @@ROWCOUNT = 1 AND [Id] = scope_identity();
+dbug: 9/29/2022 11:43:09.297 RelationalEventId.TransactionCommitted[20202] (Microsoft.EntityFrameworkCore.Database.Transaction)
+      Committed transaction.
+```
+
+EF7 detects that the transaction is not needed here and so removes these calls:
+
+```output
+info: 9/29/2022 11:42:34.776 RelationalEventId.CommandExecuted[20101] (Microsoft.EntityFrameworkCore.Database.Command)
+      Executed DbCommand (25ms) [Parameters=[@p0='MyBlog' (Nullable = false) (Size = 4000)], CommandType='Text', CommandTimeout='30']
+      SET IMPLICIT_TRANSACTIONS OFF;
+      SET NOCOUNT ON;
+      INSERT INTO [Blogs] ([Name])
+      OUTPUT INSERTED.[Id]
+      VALUES (@p0);
+```
+
+This removes two database roundtrips, which can make a huge difference to overall performance, especially when the latency of calls to the database is high. In typical production systems, the database is not co-located on the same machine as the application. This means latency is often relatively high, making this optimization particularly effective in real-world production systems.
+
+### Improved SQL for simple Identity insert
+
+The case above inserts a single row with an `IDENTITY` key column and no other database-generated values. EF7 simplifies the SQL in this case by using `OUTPUT INSERTED`. While this simplification is not valid for many other cases, it is still significant because this kind of single-row insert is very common in many applications.
```

Comment:
Yeah, just note that I haven't measured this in isolation, and I suspect that the difference between a single INSERT with `OUTPUT INSERTED` or an INSERT followed by a batched `SELECT SCOPE_IDENTITY` would be *very* similar. The improvement here really is more about unlocking the removal of the transaction (1 statement instead of 2).

(technically, you don't need an insert around INSERT followed by `SELECT SCOPE_IDENTITY`, since the latter select just reads from the connection state and doesn't modify or even read the database. But once you have a non-identity database-generated column things become more complicated...)